pkg/kubectl/cmd/apply/apply.go

```diff
@@ -398,43 +398,49 @@ func (o *ApplyOptions) Run() error {
 			if err != nil {
 				return cmdutil.AddSourceToErr("serverside-apply", info.Source, err)
 			}
+			
 			options := metav1.PatchOptions{
 				Force:        &o.ForceConflicts,
 				FieldManager: o.FieldManager,
 			}
 			if o.ServerDryRun {
 				options.DryRun = []string{metav1.DryRunAll}
 			}
+
 			obj, err := resource.NewHelper(info.Client, info.Mapping).Patch(
 				info.Namespace,
 				info.Name,
 				types.ApplyPatchType,
 				data,
 				&options,
 			)
-			if err == nil {
-				info.Refresh(obj, true)
-				metadata, err := meta.Accessor(info.Object)
-				if err != nil {
-					return err
-				}
-				visitedUids.Insert(string(metadata.GetUID()))
-				count++
-				if len(output) > 0 && !shortOutput {
-					objs = append(objs, info.Object)
-					return nil
-				}
-				printer, err := o.ToPrinter("serverside-applied")
-				if err != nil {
-					return err
+			if err != nil {
+				if isIncompatibleServerError(err) {
```

Comment:
I don't know if we still want to special-case this. We could just fail with the given error (which is probably `unsupported media type`), or return a completely different error. But I'm not super convinced by the `klog.Warning`.

Reply:
You could try this maybe:

```
if isIncompatibleServerError(err) {
  err = fmt.Errorf("Server-side apply not enabled (%v)", err)
}
return err
```

Reply:
I updated this. The only change was slight wording of the error message

Reply:
Awesome thanks